pkg/ui/react-app/src/pages/graph/ExpressionInput.tsx

```diff
@@ -88,11 +94,39 @@ const ExpressionInput: FC<PathPrefixProps & CMExpressionInputProps> = ({
   enableAutocomplete,
   enableHighlighting,
   enableLinter,
+  getExplain,
+  type,
 }) => {
   const containerRef = useRef<HTMLDivElement>(null);
   const viewRef = useRef<EditorView | null>(null);
   const { theme } = useTheme();
-
+  const handleExplain = async (query: string) => {
+    try {
+      const queryParams = new URLSearchParams({
+        query,
+        engine: 'thanos',
```

Comment:
I don't think this is correct. We need to pass the same parameters as query endpoints, to the explain endpoints.

I think the best way would be to do similar to existing. So passing a function as a prop to `ExpressionInput` component called `executeExplain` which would call the explain endpoints and pass the same params as query. The `executeExplain` function can be defined in `Panel.tsx` in the same format. 🙂